src/app/tide/peptide_mods3.cc

```diff
@@ -61,275 +61,251 @@ class IModsOutputter {
-    } else if (FLAGS_min_mods < 1) {
-      WritePeptide(peptide); // write unmodified peptide
-    }
-    if (maxMods_ == 0) {
-      return;
-    }
-
-    ResultMods resultMods(modTable_, modMaxCounts_, maxMods_, peptide, proteins_);
-    double unmodifiedMass = peptide->mass();
+  uint64_t Total() const { return totalWritten_; }
 
-    while (resultMods.Next()) {
-      resultMods.ModifyPeptide();
-      WritePeptide(peptide); 
-      if (totalWritten_ % 10000000 == 0) {
-        carp(CARP_INFO, "Wrote %lu modified target peptides to temp files", totalWritten_);
+  void InitCountsMapper() {
+ 
+    const vector<double>& deltas = *mod_table_->OriginalDeltas();
+    delta_by_file_.resize(numFiles_);
+    for (int i = 0; i < numFiles_; ++i) {
+      double total_delta = 0;
+      int x = i;
+      for (int j = max_counts_.size() - 1; j >= 0; --j) {
+        int digit = x / counts_mapper_vec_[j];
+        x %= counts_mapper_vec_[j];
+        total_delta += deltas[j] * digit;
       }
+      delta_by_file_[i] = total_delta;
     }
   }
 
-  // Return the total number of peptides written
-  uint64_t Total() const { return totalWritten_; }
+  void Output(pb::Peptide* peptide) {
+    peptide_ = peptide;
+    const pb::Location& loc = peptide->first_location();
+    residues_ = proteins_[loc.protein_id()]->residues().data() + loc.pos();
+    vector<int> counts(max_counts_.size(), 0);
+    OutputNtermMods(0, counts);
+  }
 
  private:
+  string tempDir_;
+  int numFiles_;
+  int64_t modPeptideCnt_;
+  
   unsigned long long memory_limit_;
   vector<pb::Peptide> pb_peptide_list_;
   unsigned long long temp_file_cnt_;
   vector<string> temp_file_names_;
-  class ResultMods {
-   private:
-    class ModState { // mod state for a single residue
-     public:
-      ModState(vector<int>::const_iterator begin, vector<int>::const_iterator end)
-        : begin_(begin), end_(end), current_(begin) {}
-      vector<int>::const_iterator begin_, end_, current_;
-    };
-
-    bool NextCombination() {
-      do {
-        if (modifiedIndices_.Size() < FLAGS_min_mods) {
-          modifiedIndices_ = MathUtil::Combination(modifiedIndices_.N(), FLAGS_min_mods);
-        } else if (modifiedIndices_.HasNext()) {
-          modifiedIndices_.Next();
-        } else {
-          size_t newK = modifiedIndices_.Size() + 1;
-          if (newK > maxMods_ || newK > modifiedIndices_.N()) {
-            return false;
-          }
-          modifiedIndices_ = MathUtil::Combination(modifiedIndices_.N(), newK);
-        }
-        // check that all residues in this combination can be modified
-        bool valid = true;
-        const vector<size_t>& modIndices = modifiedIndices_.Values();
-        for (vector<size_t>::const_iterator i = modIndices.begin(); i != modIndices.end(); i++) {
-          if (modIndexStates_[*i].empty()) {
-            valid = false;
-            break;
+  uint64_t totalWritten_;
+
+  //terminal modifications count as a modification and hence
```

Comment:
add space after "//" and before first character in this comment and next comment.

Reply:
Done.